hazelcast/src/main/java/com/hazelcast/jet/impl/MasterJobContext.java

```diff
@@ -1044,7 +1091,8 @@ private void completeWithMetrics(CompletableFuture<List<RawJobMetrics>> clientFu
         if (firstThrowable != null) {
             clientFuture.completeExceptionally(firstThrowable);
         } else {
-            clientFuture.complete(toList(metrics, e -> (RawJobMetrics) e.getValue()));
+            setExecutionMetrics(toList(metrics, e -> (RawJobMetrics) e.getValue()));
```

Comment:
is there a race condition possible? `completeWithMetrics` is invoked on demand from `getJobMetrics` (`Job.getMetrics()`). What if during such invocation the job finishes? Will we overwrite terminal metrics with older ones?

Reply:
7fd39226f75bf121591b561b1ed17c513af7247a

Reply:
better fix in feecf8dcea870b9ddf9a607726cc0c7073939dc9